src/main/flight/gps_rescue.c

```diff
@@ -953,12 +939,13 @@ float gpsRescueGetImuYawCogGain(void)
 
 float gpsRescueGetThrottle(void)
 {
-    // Calculated a desired commanded throttle scaled from 0.0 to 1.0 for use in the mixer.
-    // We need to compensate for min_check since the throttle value set by gps rescue
-    // is based on the raw rcCommand value commanded by the pilot.
+    // Calculate the commanded throttle scaled from 0.0 to 1.0 for use in the mixer.
+    // minCheck can be less than, or greater than, PWM_RANGE_MIN, but is usually at default of 1050
+    // it is the value at which the user expects the motors to start spinning (leaving a deadband from 1000 to 1050)
+    // rescue throttle min can't be less than gps_rescue_throttle_min (1100) or greater than max (1750)
+    // we scale throttle from mincheck to PWM_RANGE_MAX when mincheck is greater than PWM_RANGE_MIN, otherwise from PWM_RANGE_MIN to PWM_RANGE_MAX
     float commandedThrottle = scaleRangef(rescueThrottle, MAX(rxConfig()->mincheck, PWM_RANGE_MIN), PWM_RANGE_MAX, 0.0f, 1.0f);
-    commandedThrottle = constrainf(commandedThrottle, 0.0f, 1.0f);
```

Comment:
(you can configure limits to get negative value here)

Reply:
Yes.  Thanks.

I'll run through the permutations in case there is a better solution.

- If mincheck is less than 1000, the lower input comparison will be to PWM_RANGE_MIN, of 1000.
- hence the input range cannot be wider than 1000-2000, and the upper 2000 is fixed.
- however if mincheck is greater than 1000, the input range could be narrower, eg if mincheck is 1200, then the input range would be 1200-2000, and any incoming value less than 1200 would return a negative value. And most of us set min_check to a value above 1000; its default is 1050.
- the incoming rescueThrottle value is scaled between the settings for GPS Rescue throttleMin and GPS Rescue throttleMax. gpsRescueConfig()->throttleMin can't be set less than 1100.

I think - if I follow it correctly - that we could get a negative result if min_check is configured higher than usual; eg above gpsRescueConfig()->throttleMin.  This is unusual in DShot but could occur in analog situations where the motors don't start spinning until a relatively high value.

Currently the GPS Rescue, the minimum and hover throttle values don't consider min_check.  So if the user moves min_check upwards, it will affect their throttle stick position to motor output response.  And the scaling will move the GPS Rescue throttle value upwards.  I think that we wouldn't want that; we would want the GPS Rescue code to be sent the same value, we don't care about the user's deadband at the bottom of the stick, we care about the GPS unit behaving consistently.  And what matters to the GPS code is, where do the motors 'start spinning' ?  And that value relates to min_command.  Unfortunately mincommand can be set above 1000 also.

My other throttle related PR sets the idle value to some percentage above min_command.

What should really happen for the throttle stick response is that min_check should always be offset some amount above min_command.  It should not be possible to configure min_check *less than* min_command. Min_check is essentially a kind of 'throttle stick deadband' where min_command is sent until min_check is exceeded, and then throttle is scaled 0-1 while the stick is above min_check. 

So what we should really do, is remove min_check, and make it a percentage of some kind.  Our current default of min_check at 1050 is set to 5% above our default min_command.  So we could change min_check to be a value of 50 in this case, meaning 'add 5%', and apply it as a percentage above min_command.

That would solve this mess properly.  It would no longer be possible to set min_command below min_check.  We could then scale these things a lot more easily and safely.

The other way to do this would be to do all the throttle calculations in a throttle range 0-1, rather than in a 1000-2000 motor units range.  But that also should best be done later in a separate PR.

OK.

For now, I'll put that constraint back, and later I'll look at a PR that configures min_check as a percentage above min_command.  I expect that to make a lot of things easier since we would then scale GPS Rescue and similar things from the min_check value (where motors should spin) and the output range from min_command.